scalding-parquet/src/test/scala/com/twitter/scalding/parquet/tuple/macros/MacroUnitTests.scala

```diff
@@ -0,0 +1,86 @@
+package com.twitter.scalding.parquet.tuple.macros
+
+import org.scalatest.{ WordSpec, Matchers }
+import parquet.schema.MessageTypeParser
+
+case class SampleClassA(x: Int, y: String)
+case class SampleClassB(a1: SampleClassA, a2: SampleClassA, y: String)
+case class SampleClassC(a: SampleClassA, b: SampleClassB, c: SampleClassA, d: SampleClassB, e: SampleClassB)
+case class SampleClassD(a: String, b: Boolean, c: Option[Short], d: Int, e: Long, f: Float, g: Option[Double])
+
+class MacroUnitTests extends WordSpec with Matchers {
+
+  "MacroGenerated Parquet schema generator" should {
+
+    "Generate parquet schema for SampleClassA" in {
+      val schema = Macros.caseClassParquetSchema[SampleClassA]
+      val expectedSchema = MessageTypeParser.parseMessageType("""
+          |message SampleClassA {
+          |  required int32 x;
+          |  required binary y;
+          |}
+        """.stripMargin)
+      schema shouldEqual expectedSchema
+    }
+
+    "Generate parquet schema for SampleClassB" in {
+      val schema = Macros.caseClassParquetSchema[SampleClassB]
+
+      val expectedSchema = MessageTypeParser.parseMessageType("""
+        |message SampleClassB {
+        |  required int32 a1.x;
+        |  required binary a1.y;
+        |  required int32 a2.x;
+        |  required binary a2.y;
+        |  required binary y;
+        |}
+      """.stripMargin)
+      schema shouldEqual expectedSchema
+    }
+
+    "Generate parquet schema for SampleClassC" in {
+      val schema = Macros.caseClassParquetSchema[SampleClassC]
+
+      val expectedSchema = MessageTypeParser.parseMessageType("""
+        |message SampleClassC {
+        |  required int32 a.x;
+        |  required binary a.y;
+        |  required int32 b.a1.x;
+        |  required binary b.a1.y;
+        |  required int32 b.a2.x;
+        |  required binary b.a2.y;
+        |  required binary b.y;
+        |  required int32 c.x;
+        |  required binary c.y;
+        |  required int32 d.a1.x;
+        |  required binary d.a1.y;
+        |  required int32 d.a2.x;
+        |  required binary d.a2.y;
+        |  required binary d.y;
+        |  required int32 e.a1.x;
+        |  required binary e.a1.y;
+        |  required int32 e.a2.x;
+        |  required binary e.a2.y;
+        |  required binary e.y;
```

Comment:
@julienledem Hi Julien, totally agree with you, it will be better to use nesting groups. The reason I make the generated schema flat is because today [TupleWriteSupport](https://github.com/Parquet/parquet-mr/blob/master/parquet-cascading/src/main/java/parquet/cascading/TupleWriteSupport.java) only supports primitive type for right now. 

```
     if (field.isPrimitive()) {
        writePrimitive(record, field.asPrimitiveType());
      } else {
        throw new UnsupportedOperationException("Complex type not implemented");
      }
```